### PR TITLE
Split cohttp + switch to jbuilder

### PR DESCRIPTION
**Pull request by rgrinberg**
Cohttp is split into the following opam subpackages:

* cohttp
* cohttp-async
* cohttp-lwt
* cohttp-lwt-jsoo
* cohttp-lwt-unix

Things left to do:
- [x] update synopsis in jbuild files
- [x] Include cohttp version in build
- [x] update dependencies. For example we need jsoo >= 3.0 now.
- [x] Implement META compatibility the way jsoo does it.
- [x] Fix the jsoo build
- [x]  Remove oasis
- [x] travis

### CODE REVIEW

File: lib/jbuild [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+(library
+ ((name cohttp)
+  (public_name cohttp)
+  (preprocess (pps (ppx_fields_conv ppx_sexp_conv)))
+  (libraries (re.emacs
+              stringext
+              uri
+              fieldslib
+              sexplib
+              ppx_fields_conv
```
Comment (diml) [inline lib/jbuild]: this is surprising, I don't suppose cohttp uses the the ppx rewriter itself at runtime?

**Comment by rgrinberg** [comment]
No, it doesn't. Thanks for pointing it out

**Comment by rgrinberg** [comment]
@avsm @dsheets @samoht I intend to merge and release this after jsoo > 3.0 is out.

**Comment by rgrinberg** [comment]
build-test and build-doc will come once jbuilder supports it

**Comment by vbmithr** [comment]
Sorry I accidentally pushed in this branch. But I think my changes are good, so I leave you review them!

**Comment by rgrinberg** [comment]
Not at all. Feel free to help out.

**Comment by rgrinberg** [comment]
ASAP will have to wait until jsoo is released unfortunately. This branch requires jsoo's 3.0's omp based ppx

File: cohttp-lwt-jsoo.opam [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  "lwt"
+  "js_of_ocaml"
+  "js_of_ocaml-ppx"
+  "js_of_ocaml-lwt"
+]
+conflicts: [
+  "lwt" {< "2.5.0"}
+  "js_of_ocaml" {< "2.8"}
+]
+available: [ocaml-version >= "4.01.0"]
```
Comment (avsm) [inline cohttp-lwt-jsoo.opam]: I think it might be time to bump to 4.02.3 as our minimum version

**Comment by rgrinberg** [comment]
@avsm fixed the minimum OCaml versions.

Would you mind taking care of travis for this PR as well?

**Comment by rgrinberg** [comment]
The only reason this is failing is because of jsoo 3.0. I will just merge this and release after jsoo 3.0

**Pull request merged**